internal/config/config.go

```diff
@@ -723,6 +734,34 @@ func (c AzureSEVSNP) GetMeasurements() measurements.M {
 	return c.Measurements
 }
 
+// SetMeasurements updates a config's measurements using the given measurements.
+func (c *AzureSEVSNP) SetMeasurements(m measurements.M) {
+	c.Measurements = m
+}
+
+// NewerThan returns true if the config is newer than the given config.
+func (c AzureSEVSNP) NewerThan(old AttestationCfg) (bool, error) {
+	otherCfg, ok := old.(*AzureSEVSNP)
+	if !ok {
+		return false, fmt.Errorf("cannot compare %T with %T", c, old)
+	}
+
+	if c.FirmwareSignerConfig.EnforcementPolicy == idkeydigest.MAAFallback &&
+		otherCfg.FirmwareSignerConfig.EnforcementPolicy != idkeydigest.MAAFallback {
+		return false, errors.New("cannot update enforcement policy to 'maaFallback'")
+	}
+
+	acceptedKeyDigestsNewer := !c.FirmwareSignerConfig.AcceptedKeyDigests.EqualTo(otherCfg.FirmwareSignerConfig.AcceptedKeyDigests)
+	measurementsNewer := !c.Measurements.EqualTo(otherCfg.Measurements)
+	bootloaderNewer := c.BootloaderVersion > otherCfg.BootloaderVersion
+	teeNewer := c.TEEVersion > otherCfg.TEEVersion
+	snpNewer := c.SNPVersion > otherCfg.SNPVersion
+	microcodeNewer := c.MicrocodeVersion > otherCfg.MicrocodeVersion
+	rootKeyNewer := !bytes.Equal(c.AMDRootKey.Raw, otherCfg.AMDRootKey.Raw)
+
+	return acceptedKeyDigestsNewer || measurementsNewer || bootloaderNewer || teeNewer || snpNewer || microcodeNewer || rootKeyNewer, nil
+}
+
 // SNPFirmwareSignerConfig is the configuration for validating the firmware signer.
 type SNPFirmwareSignerConfig struct {
 	// description: |
```

Comment:
Not really the line I want to comment on. In line 729: maybe move `\nValues are enforced according to`... down to the `EnforcementPolicy`?

Reply:
See https://github.com/edgelesssys/constellation/pull/1561#discussion_r1158499793